src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs

```diff
@@ -43,7 +43,10 @@ public Type FindStartupType(string startupAssemblyName, IList<string> diagnostic
             var environmentName = _hostingEnv.EnvironmentName;
             if (string.IsNullOrEmpty(startupAssemblyName))
             {
-                throw new ArgumentException("Value cannot be null or empty.", nameof(startupAssemblyName));
+                throw new ArgumentException(
+                    string.Format("A startup method, startup type or startup assembly is required. If specifying an assembly, '{0}' cannot be null or empty.", 
+                    nameof(startupAssemblyName)), 
+                    nameof(startupAssemblyName));
```

Comment:
The first one is for the format string, the second is for the parameter name of the exception. I don't think the second one is that useful but it was there previously so I didn't remove it.